server/base/src/main/java/org/apache/accumulo/server/master/recovery/RecoveryPath.java

```diff
@@ -50,7 +47,8 @@ public static Path getRecoveryPath(Path walPath, String portHost) {
 
       // drop wal
       walPath = walPath.getParent();
-      walPath = new Path(walPath, FileType.RECOVERY.getDirectory() + "-" + convertToSha1(portHost));
+      walPath =
+          new Path(walPath, FileType.RECOVERY.getDirectory() + "-" + DigestUtils.sha1Hex(portHost));
```

Comment:
Recently it was noted that using sha1 can be flagged by some code security scan utilities - would using sha256 or even sha512 eliminate those utilities from flagging "unsecure usages"?  This is not a security related hash so it does not matter, but those utilities don't take that into account.

The length of the string could also be shortened using something like:

```String s = new BigInteger(DigestUtils.sha256(portHost).toString(MAX_RADIX).substring(0,40)```

This would have the same number of characters as the sha1 digest, but more "bits" using base36 vs base16 - or the string length could be shortened so that it was equivalent to the 40 digit sha1 without increasing the (small) risk of a collision. This also would allow using sha512 without a ridiculous string length.

Reply:
I believe there's a variant of the sha256 method in DigestUtils that returns a hex-encoded string, that can simply be truncated directly, rather than using BigInteger, which shouldn't be necessary.

However, I'm not sure that this hashing of a location strategy is going to work at all, based on the conversation above in https://github.com/apache/accumulo/pull/1803#discussion_r568151689